src/message/__tests__/messageActions-test.js

```diff
@@ -3,19 +3,88 @@ import configureStore from 'redux-mock-store';
 // $FlowFixMe[untyped-import]
 import thunk from 'redux-thunk';
 
+import { combinedThunkExtras } from '../../boot/store';
 import { navigateToChat } from '../../nav/navActions';
 import * as NavigationService from '../../nav/NavigationService';
-import { doNarrow } from '../messagesActions';
-import { streamNarrow } from '../../utils/narrow';
+import { messageLinkPress, doNarrow } from '../messagesActions';
+import { streamNarrow, topicNarrow, pmNarrowFromUsersUnsafe } from '../../utils/narrow';
 import * as eg from '../../__tests__/lib/exampleData';
 import type { GlobalState } from '../../reduxTypes';
 import type { Action } from '../../actionTypes';
+import type { Stream, User } from '../../types';
+import { mock_ } from '../../__tests__/lib/intl';
+import * as logging from '../../utils/logging';
 
-const mockStore = configureStore([thunk]);
+const mockStore = configureStore([thunk.withExtraArgument(combinedThunkExtras)]);
```

Comment:
Interesting! It looks like we haven't written a new test using this `redux-mock-store` library since… 2020, in e5268bbaaa5c34956c38cb5be8d31c970ed2d150. Before that it was 2018 in 75240efbf8d0c347f71acc5405ff7250b2324575, and before that just two commits in 2017, namely b0c7323d18b053ff171d504db8fd1f067ce0fdfd and bc14f0d35afa7a52e589c9bc394ed93c311d80d7. (Based on `git log -S mockStore` as well as some other `git log` searches.) I must have reviewed the later two of those four commits, but I'd forgotten about this functionality.

Mostly that's meant we just don't have tests for the actions code; `git ls-files src/'*'Actions-test.js` is very thin.

I guess one thing that's influenced my approach is that the existing tests we've had using this, including the one small existing test in this file, have been pretty unimpressive:
https://github.com/zulip/zulip-mobile/pull/5052#issuecomment-965860359

I like these new tests, though. I think it makes a lot of sense to have an end-to-end test that, when you try to open a URL that looks like one of these, we navigate to the right spot. If anything it'd be better if it were *more* end-to-end — but that sounds like more work, so let's not do that here. :slightly_smiling_face: We do have such tests in zulip-flutter already.